networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_cfg_syncer.py

```diff
@@ -0,0 +1,207 @@
+                                       segment_nat_dict,
+                                       parsed_cfg)
+
+        invalid_cfg += self.clean_interfaces(conn,
+                                             intf_segment_dict,
+                                             segment_nat_dict,
+                                             parsed_cfg)
+
+        invalid_cfg += self.clean_vrfs(conn, router_id_dict, parsed_cfg)
+        LOG.debug("invalid_cfg = %s " % (pp.pformat(invalid_cfg)))
+        return invalid_cfg
+
+    def clean_routes(self,
+                     conn,
+                     router_id_dict,
+                     intf_segment_dict,
+                     segment_nat_dict,
+                     parsed_cfg,
+                     route_regex):
+        delete_route_list = []
+        routes = parsed_cfg.find_objects(route_regex)
+        for route in routes:
+            LOG.info(_LI("\ndefault route: %s"), (route))
+            match_obj = re.match(route_regex, route.text)
+            is_multi_region_enabled = cfg.CONF.multi_region.enable_multi_region
+            if (is_multi_region_enabled):
+                router_id, region_id, segment_id, next_hop = (
+                    match_obj.group(1, 2, 3, 4))
+                LOG.info(_LI("    router_id: %(router_id)s,"
+                             ", region_id: %(region_id)s, segment_id:"
+                             " %(segment_id)s, next_hop: %(next_hop)s") %
+                         {'router_id': router_id,
+                          'region_id': region_id,
+                          'segment_id': segment_id,
+                          'next_hop': next_hop})
+                my_region_id = cfg.CONF.multi_region.region_id
+                other_region_ids = cfg.CONF.multi_region.other_region_ids
+                if region_id != my_region_id:
+                    if region_id not in other_region_ids:
+                        delete_route_list.append(route.text)
+                    else:
+                        # skip because some other deployment owns
+                        # this configuration
+                        continue
+            else:
+                router_id, segment_id, next_hop = (
+                    match_obj.group(1, 2, 3))
+
+            LOG.info(_LI("    router_id: %(router_id)s, segment_id:"
+                         " %(segment_id)s, next_hop: %(next_hop)s") %
+                     {'router_id': router_id,
+                      'segment_id': segment_id,
+                      'next_hop': next_hop})
+
+            # Check that VRF exists in openstack DB info
+            if router_id not in router_id_dict:
+                LOG.info(_LI("router not found for route, deleting"))
+                delete_route_list.append(route.text)
+                continue
+
+            # Check that router has external network and segment_id matches
+            router = router_id_dict[router_id]
+            if "gw_port" not in router:
```

Comment:
since you also call gw_port = router['gw_port'] at line#181, I think its more efficient to do things like this here:
gw_port = router.get('gw_port')
if gw_port is None:
   XXXXX
   continue